File: QEasyWidgets/Components/ComboBox.py
from PySide6.QtGui import *
from PySide6.QtCore import *
from PySide6.QtWidgets import *

from ..Common.QFunctions import *
from ..Resources.Sources import *

##############################################################################################################################

class ComboBoxBase(QComboBox):
    '''
    '''
    def __init__(self,
        parent: Optional[QWidget] = None
    ):
        super().__init__(parent)

        self.setFocusPolicy(Qt.StrongFocus)

        StyleSheetBase.ComboBox.Apply(self)

    def wheelEvent(self, event: QWheelEvent) -> None:
        event.ignore()

    def ClearDefaultStyleSheet(self) -> None:
        StyleSheetBase.ComboBox.Deregistrate(self)

##############################################################################################################################

File: c++/QEasyWidgets/Components/ComboBox.cpp
#include "ComboBox.h"

#include "../Common/StyleSheet.h"


namespace QEW {

ComboBoxBase::ComboBoxBase(QWidget *parent)
    : QComboBox(parent)
{
    init();
}

void ComboBoxBase::init()
{
    setFocusPolicy(Qt::StrongFocus);
    StyleSheetBase::apply(this, StyleSheetBase::ComboBox);
}

void ComboBoxBase::wheelEvent(QWheelEvent *event)
{
    event->ignore();
}

void ComboBoxBase::setBorderless(bool borderless)
{
    setProperty("isBorderless", borderless);
}

void ComboBoxBase::setTransparent(bool transparent)
{
    setProperty("isTransparent", transparent);
}

void ComboBoxBase::clearDefaultStyleSheet()
{
    // TODO: Implement deregistration if StyleSheetBase supports it
}

} // namespace QEW
